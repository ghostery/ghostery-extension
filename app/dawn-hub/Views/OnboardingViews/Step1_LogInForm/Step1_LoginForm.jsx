/**
 * Browser Log In Form
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2021 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import ClassNames from 'classnames';

/**
 * A Functional React component for rendering the Log In Form
 * @return {JSX} JSX for rendering the Log In Form of the Dawn Hub app
 * @memberof DawnHubViews
 */
const Step1_LogInForm = (props) => {
	const {
		email,
		password,
		emailError,
		passwordError,
		handleSubmit,
		handleInputChange,
		handleForgotPassword,
	} = props;

	const emailInputClassNames = ClassNames('Step1_LogInForm__inputBox', {
		error: emailError,
	});
	const passwordInputClassNames = ClassNames('Step1_LogInForm__inputBox', {
		error: passwordError,
	});

	return (
		<form className="Step1_LogInForm" onSubmit={handleSubmit}>
			<div className="Step1_LogInForm__item row align-center-middle">
				<div className="columns small-12">
					<label htmlFor="login-email" className="Step1_LogInForm__inputLabel">
						{t('email_colon')}
					</label>
					<input
						id="login-email"
						className={emailInputClassNames}
						name="email"
						type="text"
						value={email}
						onChange={handleInputChange}
						pattern=".{1,}"
						autoComplete="off"
						placeholder="example@mail.com"
					/>
					{emailError && (
						<div className="Step1_LogInForm__inputError">
							{t('please_enter_a_valid_email')}
						</div>
					)}
				</div>
			</div>
			<div className="Step1_LogInForm__item row align-center-middle">
				<div className="columns small-12">
					<label htmlFor="login-password" className="Step1_LogInForm__inputLabel">
						{t('password_colon')}
					</label>
					<input
						id="login-password"
						className={passwordInputClassNames}
						name="password"
						type="password"
						value={password}
						onChange={handleInputChange}
						pattern=".{1,}"
					/>
					{passwordError && (
						<div className="Step1_LogInForm__inputError">
							{t('hub_login_label_password_invalid')}
						</div>
					)}
				</div>
			</div>
			<div className="Step1_LogInForm__item row align-center-middle">
				<div className="columns small-12">
					<span className="Step1_LogInForm__link">
						<div className="Step1_LogInForm__forgotPassword" onClick={handleForgotPassword}>
							{t('forgot_password')}
						</div>
					</span>
				</div>
			</div>
			<div className="Step1_LogInForm__ctaButtonContainer">
				<button type="submit" className="Step1_LogInForm__ctaButton">{t('sign_in')}</button>
			</div>
		</form>
	);
};

// PropTypes ensure we pass required props of the correct type
Step1_LogInForm.propTypes = {
	email: PropTypes.string.isRequired,
	password: PropTypes.string.isRequired,
	emailError: PropTypes.bool.isRequired,
	passwordError: PropTypes.bool.isRequired,
	handleSubmit: PropTypes.func.isRequired,
	handleInputChange: PropTypes.func.isRequired,
};

export default Step1_LogInForm;
