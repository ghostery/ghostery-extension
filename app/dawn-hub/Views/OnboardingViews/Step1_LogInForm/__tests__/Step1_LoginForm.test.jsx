/**
 * Create Account View Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2021 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router';
import Step1_LoginForm from '../Step1_LoginForm';

const noop = () => {};
describe('app/hub/Views/Step1_LoginForm component', () => {
	const initialState = {
		email: '',
		password: '',
		emailError: false,
		passwordError: false,
		handleSubmit: noop,
		handleInputChange: noop,
		handleForgotPassword: noop
	};
	describe('Snapshot tests with react-test-renderer', () => {
		test('Login Form view is rendered correctly', () => {

			const component = renderer.create(
				<MemoryRouter>
					<Step1_LoginForm {...initialState} />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('Shallow snapshot tests rendered with Enzyme', () => {
		test('the happy path of the component', () => {
			const happyState = {
				...initialState,
				email: 'test@example.com',
				password: 'examplePassword',
				handleSubmit: jest.fn(),
			}

			const component = shallow(<Step1_LoginForm {...happyState} />);
			expect(happyState.handleSubmit.mock.calls.length).toBe(0);
			component.find('form').simulate('submit');
			expect(happyState.handleSubmit.mock.calls.length).toBe(1);
		});

		test('the sad path of the component with errors', () => {
			const sadState = {
				...initialState,
				email: 'test@example.com',
				password: 'examplePassword',
				emailError: true,
				passwordError: true,
				handleSubmit: jest.fn(),
			};

			const component = shallow(<Step1_LoginForm {...sadState} />);

			expect(sadState.handleSubmit.mock.calls.length).toBe(0);
			component.find('form').simulate('submit');
			expect(sadState.handleSubmit.mock.calls.length).toBe(1);
		})
	});
});
