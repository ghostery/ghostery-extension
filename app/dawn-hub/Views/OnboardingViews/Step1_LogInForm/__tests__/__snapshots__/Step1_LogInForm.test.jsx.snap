// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/hub/Views/Step1_LoginForm component Snapshot tests with react-test-renderer Login Form view is rendered correctly 1`] = `
<form
  className="Step1_LogInForm"
  onSubmit={[Function]}
>
  <div
    className="Step1_LogInForm__item row align-center-middle"
  >
    <div
      className="columns small-12"
    >
      <label
        className="Step1_LogInForm__inputLabel"
        htmlFor="login-email"
      >
        email_colon
      </label>
      <input
        autoComplete="off"
        className="Step1_LogInForm__inputBox"
        id="login-email"
        name="email"
        onChange={[Function]}
        pattern=".{1,}"
        placeholder="example@mail.com"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="Step1_LogInForm__item row align-center-middle"
  >
    <div
      className="columns small-12"
    >
      <label
        className="Step1_LogInForm__inputLabel"
        htmlFor="login-password"
      >
        password_colon
      </label>
      <input
        className="Step1_LogInForm__inputBox"
        id="login-password"
        name="password"
        onChange={[Function]}
        pattern=".{1,}"
        type="password"
        value=""
      />
    </div>
  </div>
  <div
    className="Step1_LogInForm__item row align-center-middle"
  >
    <div
      className="columns small-12"
    >
      <span
        className="Step1_LogInForm__link"
      >
        <div
          className="Step1_LogInForm__forgotPassword"
          onClick={[Function]}
        >
          forgot_password
        </div>
      </span>
    </div>
  </div>
  <div
    className="Step1_LogInForm__ctaButtonContainer"
  >
    <button
      className="Step1_LogInForm__ctaButton"
      type="submit"
    >
      sign_in
    </button>
  </div>
</form>
`;
